src/openforms/analytics_tools/models.py

```diff
@@ -0,0 +1,308 @@
+        ),
+    )
+
+    enable_siteimprove_analytics = models.BooleanField(
+        _("enable siteImprove analytics"),
+        default=False,
+        help_text=_("Enabling this installs SiteImprove"),
+    )
+
+    analytics_cookie_consent_group = models.ForeignKey(
+        "cookie_consent.CookieGroup",
+        on_delete=models.SET_NULL,
+        null=True,
+        blank=True,
+        help_text=_(
+            "The cookie group used for analytical cookies. The analytics scripts are "
+            "loaded only if this cookie group is accepted by the end-user."
+        ),
+    )
+
+    class Meta:
+        verbose_name = _("Analytics tools configuration")
+
+    @property
+    def is_matomo_enabled(self) -> bool:
+        return (
+            self.matomo_url
+            and self.matomo_site_id
+            and self.enable_matomo_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    @property
+    def is_piwik_enabled(self) -> bool:
+        return (
+            self.piwik_url
+            and self.piwik_site_id
+            and self.enable_piwik_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    @property
+    def is_piwik_pro_enabled(self) -> bool:
+        return (
+            self.piwik_pro_url
+            and self.piwik_pro_site_id
+            and self.enable_piwik_pro_site_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    @property
+    def is_siteimprove_enabled(self) -> bool:
+        return (
+            self.siteimprove_id
+            and self.enable_siteimprove_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    @property
+    def is_google_analytics_enabled(self) -> bool:
+        return (
+            self.ga_code
+            and self.gtm_code
+            and self.enable_google_analytics
+            and self.analytics_cookie_consent_group
+        )
+
+    def save(self, *args, **kwargs):
+        # If instance is being created, we can't find original values
+        if self._state.adding:
+            return super().save(*args, **kwargs)
+        original_object = self.__class__.objects.get(pk=self.pk)
+        # For each analytics provider, we check if :
+        # - the value of the provider's boolean field is changing
+        if original_object.enable_google_analytics != self.enable_google_analytics:
+            self.update_analytics_tool(
+                AnalyticsTools.google_analytics, self.is_google_analytics_enabled, []
+            )
+        if (
+            original_object.enable_matomo_site_analytics
+            != self.enable_matomo_site_analytics
+        ):
+            string_replacements_list = [
+                ("SITE_ID", self.matomo_site_id),
+                ("SITE_URL", self.matomo_url),
+                (
+                    "DOMAIN_HASH",
+                    lambda x: get_domain_hash(settings.ALLOWED_HOSTS[0], cookie_path=x),
```

Comment:
I'd pass the entire `cookie` object instead of just the `path` to make it bit more generic:

```suggestion
                    lambda cookie: get_domain_hash(settings.ALLOWED_HOSTS[0], cookie_path=cookie["path"]),
```